### DISCUSSION TITLE
flet alertDialog

### DISCUSSION THREAD

**Discussion by jk-2022** [flet-dev/flet#1805] [Q&A]
### Question

bonjour ! j'ai un souci avec le contrôle alertDialog de flet, j'ai ouvert un premier alertdialog qui contient un contenaire avec un bouton qui doit en principe ouvrir un autre alertdialog, ce dernier s'ouvre mais le premier bug! 

### ------------------------------------------------------

- [X] I have searched for answers to my question both in the issues and in previous discussions.

**Comment by ndonkoHenri** [comment]
Please provide some more context (code, video capture).

**Reply by jk-2022** [reply]
import flet
from flet import *

def main(page:Page):
    page.title="Alert Dialg"
    page.window_width=300
    page.window_height=500
    page.horizontal_alignment="center"
    
    def closeFirstDialog(e):
        first_dlg.open=False
        page.update()
        
    
    def showFirstModal(e):
        page.dialog=first_dlg
        first_dlg.open=True
        page.update()
        
    
    def closeSecondDialog(e):
        second_dlg.open=False
        page.update()
        
    
    def showSecondModal(e):
        page.dialog=second_dlg
        second_dlg.open=True
        page.update()
        
    first_btn=ElevatedButton("Open first modal", on_click=showFirstModal)
    second_btn=ElevatedButton("Open Second modal", on_click=showSecondModal)
        
    first_dlg=AlertDialog(
            title=Text("First Modal"),
            modal=True,
            content=second_btn,
            actions=[
                ElevatedButton('close',on_click=closeFirstDialog)
            ]
        )
    
    second_dlg=AlertDialog(
            title=Text("Second Modal"),
            content=Text("Second Modal open"),
            actions=[
                ElevatedButton('close',on_click=closeSecondDialog)
            ]
        )
    
        

    
    
    page.add(first_btn)
    page.update()
    

flet.app(main)

**Reply by FeodorFitsner** [reply]
Try adding some delay, like 0.1 seconds between closing first dialog and opening the second?

**Comment by jk-2022** [comment]
![nlb5oi1c](https://github.com/flet-dev/flet/assets/117345274/61fc0014-4d80-44aa-b6bd-1a220e88f62f)
![yv4syqyy](https://github.com/flet-dev/flet/assets/117345274/bfbb3143-561a-4258-b06c-a1aafbf27626)
![n7f9a6pk](https://github.com/flet-dev/flet/assets/117345274/75e84be7-a972-46a2-8097-df6af349b26f)

**Comment by ndonkoHenri** [comment]
As suggested by @FeodorFitsner, I added a small time delay/sleep just before opening the new/second dialog as seen in the modified `showSecondModal` function below: 

```python
    def showSecondModal(e):
        # close the first dialog, update the page, sleep... 
        first_dlg.open = False
        page.update()
        time.sleep(0.01)
       
        # ... before opening the second dialog
        page.dialog = second_dlg
        second_dlg.open = True
        page.update()
```
**Little Notes:** 
- I must mention that extremely small values such as `0.001` seconds failed on my end, hence I used `0.01` (any value above this will equally be OK), which is also quite good, as the delay/sleep is not noticed by the human eye :)
- You might find these utility functions useful: [`page.close_dialog()`](https://flet.dev/docs/controls/page#close_dialog), [`page.show_dialog(dialog)`](https://flet.dev/docs/controls/page#show_dialogdialog-alertdialog)

Let me know if you have any questions.

**Reply by jk-2022** [reply]
tanks😎😎😎🎉
